src/main/kotlin/lotto/domain/Lotto.kt

```diff
@@ -0,0 +1,12 @@
+package lotto.domain
+
+import lotto.domain.generator.AutoLottoGenerator
+import lotto.domain.generator.LottoGenerator
+
+data class Lotto(private val generator: LottoGenerator = AutoLottoGenerator) {
+    private val numbers = generator.execute().sorted()
+
+    override fun toString() = "[${numbers.joinToString()}]"
+
+    fun getMatchCount(other: Lotto) = (numbers + other.numbers).groupingBy { it }.eachCount().count { it.value > 1 }
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
    fun getMatchCount(other: Lotto) = numbers.count { other.numbers.contains(it) }
```